Geode.Angular/src/app/interceptors/auth.interceptor.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { ACCESS_TOKEN_KEY } from '../shared/constants/storages.constants';
+import {
+  AUTH_RULE_HEADER_NAME,
+  AUTH_RULE_HEADER_VALUES,
+} from '../shared/constants/auth-rule-header.constants';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  intercept(
+    req: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    if (
+      req.headers.get(AUTH_RULE_HEADER_NAME) === AUTH_RULE_HEADER_VALUES.APPLY
```

Comment:
Better use of reverse side, you should  **skip** the Auth header if I set the header to SKIP, because you will use this header on 90% of ur APIs and now you have to write APPLY on each of them